src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletManager.cs

```diff
@@ -1011,24 +1014,25 @@ public void EnableFederation(string password, string mnemonic = null, string pas
                 }
 
                 // Create a wallet file.
-                string encryptedSeed = extendedKey.PrivateKey.GetEncryptedBitcoinSecret(password, this.network).ToWif();
-
-                this.Wallet.EncryptedSeed = encryptedSeed;
-                this.SaveWallet();
+                key = extendedKey.PrivateKey;
+                encryptedSeed = key.GetEncryptedBitcoinSecret(password, this.network).ToWif();
             }
 
-            if (onlyImportKey)
-                return;
-
             try
             {
-                Key key = Key.Parse(this.Wallet.EncryptedSeed, password, this.Wallet.Network);
+                if (key == null)
+                    key = Key.Parse(encryptedSeed, password, this.Wallet.Network);
 
-                this.Secret = new WalletSecret() { WalletPassword = password };
                 this.isFederationActive = key.PubKey.ToHex() == this.federationGatewaySettings.PublicKey;
```

Comment:
We should probably just do a `this.isFederationActive = true;` after the saveWallet, right at the end.
This should be called bool isValidKey or something like that.

Reply:
Done.